File: frontend/lib/domain/entities/attendance.dart
import 'package:intl/intl.dart';

class AttendanceEntity {
  final String id;
  final String studentId;
  final String subjectId;
  final DateTime date;
  final String status; // 'present', 'absent', 'holiday'

  AttendanceEntity({
    required this.id,
    required this.studentId,
    required this.subjectId,
    required this.date,
    required this.status,
  });

  String get formattedDate => DateFormat('dd MMM yyyy').format(date);
}

File: backend/app/schema/attendance.py
from pydantic import BaseModel
from datetime import date

class AttendanceBase(BaseModel):
    student_id: int
    subject_id: int
    date: date
    status: str

class AttendanceCreate(AttendanceBase):
    pass

class AttendanceOut(AttendanceBase):
    id: int

    class Config:
        orm_mode = True